File: JavaScript solutions/2545. Sort the Students by Their Kth Score.js
/**
 * @param {number[][]} score
 * @param {number} k
 * @return {number[][]}
 */
var sortTheStudents = function (score, k) {
  let res = [];
  let len = score.length;
  for (let i = 0; i < len; i++) {
    let v = score[i].slice();
    res.push([score[i][k], v]);
  }

  res.sort((a, b) => b[0] - a[0]);
  let ans = [];
  for (let i = 0; i < len; i++) {
    ans.push(res[i][1]);
  }

  return ans;
};

File: C++ solutions/2545. Sort the Students by Their Kth Score.cpp
class Solution {
public:
    vector<vector<int>> sortTheStudents(vector<vector<int>>& score, int k) {  
        vector<pair<int,vector<int>>>res;
        int len=score.size();
        for(int i=0;i<len;i++){
            vector<int>v;
            int len1=score[i].size();
            for(int j=0;j<len1;j++) v.push_back(score[i][j]);

            res.push_back(make_pair(score[i][k],v));
        }
        sort(res.begin(),res.end());
        
        vector<vector<int>>ans;
        for(int i=len-1;i>=0;i--){
            ans.push_back(res[i].second);
        }
        return ans;

    }
};